src/models/stores/persistent-ui/persistent-ui.test.ts

```diff
@@ -0,0 +1,540 @@
+        tabs: {
+          test: {
+            id: "test",
+            currentDocumentGroupId: "testSubTab1",
+            visitedDocumentGroups: {
+              testSubTab1: {
+                id: "testSubTab1",
+                currentDocumentKeys: ["doc1", "doc3"]
+              },
+              testSubTab2: {
+                id: "testSubTab2",
+                currentDocumentKeys: ["doc2"]
+              },
+              testSubTab3: {
+                id: "testSubTab3",
+                currentDocumentKeys: ["doc4"]
+              }
+            },
+          }
+        },
+        activeNavTab: "test",
+        problemWorkspace: {
+          type: "problem",
+          mode: "1-up"
+        }
+      });
+      expect(ui.currentDocumentGroupId).toBe("testSubTab1");
+
+      const tabModel = ui.activeTabModel;
+      expect(tabModel).toBeDefined();
+      if (!tabModel) throw "tabModel is undefined";
+
+      expect(tabModel.currentDocumentGroup).toBeDefined();
+      expect(tabModel.currentDocumentGroup?.primaryDocumentKey).toBe("doc1");
+      expect(tabModel.currentDocumentGroup?.secondaryDocumentKey).toBe("doc3");
+      expect(tabModel.getDocumentGroupPrimaryDocument("testSubTab2")).toBe("doc2");
+      expect([...tabModel.visitedDocumentGroups.keys()]).toEqual(["testSubTab1", "testSubTab2", "testSubTab3"]);
+    });
+    it("converts real version 1 state", () => {
+      const realV1State: PersistentUIModelV1Snapshot = {
+        "dividerPosition": 50,
+        "activeNavTab": "problems",
+        "docFilter": "Problem",
+        "primarySortBy": "Group",
+        "secondarySortBy": "None",
+        "showAnnotations": true,
+        "showTeacherContent": true,
+        "showChatPanel": true,
+        "showDocumentScroller": true,
+        "tabs": {
+            "problems": {
+                "id": "problems",
+                "openSubTab": "introduction",
+                "openDocuments": {},
+                "openSecondaryDocuments": {}
+            },
+            "teacher-guide": {
+                "id": "teacher-guide",
+                "openSubTab": "launch",
+                "openDocuments": {},
+                "openSecondaryDocuments": {}
+            },
+            "my-work": {
+                "id": "my-work",
+                "openSubTab": "Workspaces",
+                "openDocuments": {},
+                "openSecondaryDocuments": {}
+            }
+        },
+        "problemWorkspace": {
+            "type": "problem",
+            "mode": "1-up",
+            "primaryDocumentKey": "-OJdBCDm6Pq-zdpzVTPP",
+            "comparisonVisible": false,
+            "hidePrimaryForCompare": false
+        },
+        "version": "1.0.0"
+      };
+      const ui = PersistentUIModel.create({
+        version: "2.0.0",
+        tabs: {},
+        problemWorkspace: {
+          type: "problem",
+          mode: "1-up"
+        }
+      });
+      // This behavior is currently broken in MST, if that gets fixed
```

Comment:
What behavior is broken?
Will this affect actual use of the version upgrade code?

Reply:
I updated the comment to explain this better.